core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java

```diff
@@ -117,6 +117,9 @@ public static Query applyMinimumShouldMatch(BooleanQuery query, @Nullable String
         if (minimumShouldMatch == null) {
             return query;
         }
+        if (query.isCoordDisabled()) {
```

Comment:
can you leave a comment to explain that this means that this booleanquery was generated from synonyms?